packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -99,6 +187,288 @@ class ReactShallowRenderer {
+      initialState: S,
+      initialAction: A | void | null,
+    ): [S, Dispatch<A>] => {
+      this._validateCurrentlyRenderingComponent();
+      this._createWorkInProgressHook();
+      const workInProgressHook: Hook = (this._workInProgressHook: any);
+      if (this._isReRender) {
+        // This is a re-render. Apply the new render phase updates to the previous
+        // current hook.
+        const queue: UpdateQueue<A> = (workInProgressHook.queue: any);
+        const dispatch: Dispatch<A> = (queue.dispatch: any);
+        if (this._renderPhaseUpdates !== null) {
+          // Render phase updates are stored in a map of queue -> linked list
+          const firstRenderPhaseUpdate = this._renderPhaseUpdates.get(queue);
+          if (firstRenderPhaseUpdate !== undefined) {
+            (this._renderPhaseUpdates: any).delete(queue);
+            let newState = workInProgressHook.memoizedState;
+            let update = firstRenderPhaseUpdate;
+            do {
+              // Process this render phase update. We don't have to check the
+              // priority because it will always be the same as the current
+              // render's.
+              const action = update.action;
+              newState = reducer(newState, action);
+              update = update.next;
+            } while (update !== null);
+
+            workInProgressHook.memoizedState = newState;
+
+            return [newState, dispatch];
+          }
+        }
+        return [workInProgressHook.memoizedState, dispatch];
+      } else {
+        if (reducer === basicStateReducer) {
+          // Special case for `useState`.
+          if (typeof initialState === 'function') {
+            initialState = initialState();
+          }
+        } else if (initialAction !== undefined && initialAction !== null) {
+          initialState = reducer(initialState, initialAction);
+        }
+        workInProgressHook.memoizedState = initialState;
+        const queue: UpdateQueue<A> = (workInProgressHook.queue = {
+          last: null,
+          dispatch: null,
+        });
+        const dispatch: Dispatch<
+          A,
+        > = (queue.dispatch = (this._dispatchAction.bind(
+          this,
+          (this._currentlyRenderingComponent: any),
+          queue,
+        ): any));
+        return [workInProgressHook.memoizedState, dispatch];
+      }
+    };
+
+    const useState = <S>(
+      initialState: (() => S) | S,
+    ): [S, Dispatch<BasicStateAction<S>>] => {
+      return useReducer(
+        basicStateReducer,
+        // useReducer has a special case to support lazy useState initializers
+        (initialState: any),
+      );
+    };
+
+    const useMemo = <T>(
+      nextCreate: () => T,
+      inputs: Array<mixed> | void | null,
+    ): T => {
+      this._validateCurrentlyRenderingComponent();
+      this._createWorkInProgressHook();
+
+      const nextInputs =
+        inputs !== undefined && inputs !== null ? inputs : [nextCreate];
```

Comment:
To match https://github.com/facebook/react/pull/14594, let's call them `deps` (not `inputs`) and make sure we use `null` rather than `[nextCreate]` in all cases. (That's what #14594 was about.)

Reply:
I can do that, but the reason I didn't was because #14594 didn't make that change to packages/react-dom/src/server/ReactPartialRendererHooks.js.

Reply:
I think that's just an omission.